src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -141927,5 +141927,652 @@ static T GetValue2<T>(Collection c, object key)
                 //             return s2; // 4
                 Diagnostic(ErrorCode.WRN_NullReferenceReturn, "s2").WithLocation(47, 20));
         }
+
+        [Fact]
+        public void AmbigMember_DynamicDifferences()
+        {
+            var src = @"
+interface I<T> { T Item { get; } }
+
+interface I2<T> : I<T> { }
+
+interface I3 : I<dynamic>, I2<object> { }
+
+public class C
+{
+    void M(I3 i)
```

Comment:
>void M(I3 i) [](start = 4, length = 12)

I think we should also test lookup in a type parameter. I.e. when ```i``` is of type ```T```, which has ```I3``` as a constraint.

Reply:
> I think we should also test lookup in a type parameter. I.e. when i is of type T, which has I3 as a constraint.

This comment is for scenarios involving the nullability mismatch. Also, I think we should test scenarios when  constraints are just ```I<object>, I2<object>``` and ```I2<object>, I<object>```.

---
In reply to: [544360179](https://github.com/dotnet/roslyn/pull/49338#discussion_r544360179) [](ancestors = 544360179)